Kernel/KLexicalPath.cpp

```diff
@@ -29,6 +29,16 @@ bool is_canonical(StringView const& path)
     return true;
 }
 
+bool is_valid_basename(StringView const& path)
```

Comment:
As far as basenames (i.e., filenames really) are concerned "." and ".." are valid. They're just not valid for the user to specify as targets for rename. Maybe this needs a better name to accomodate for that.

Reply:
What name would you suggest? 
I thought about removing the check for '.' and '..', because the kernel should  probably just return EEXIST in that case.

Reply:
If "the old pathname names an ancestor directory of the new pathname, or either pathname argument contains a final component that is dot or dot-dot", POSIX specifies to return EINVAL. So I guess this shouldn't be checked by a method of `KLexicalPath`, but by `VFS::rename` itself.